golang/cosmos/x/swingset/genesis.go

```diff
@@ -130,36 +130,57 @@ func InitGenesis(ctx sdk.Context, k Keeper, swingStoreExportsHandler *SwingStore
 	return false
 }
 
-func ExportGenesis(ctx sdk.Context, k Keeper, swingStoreExportsHandler *SwingStoreExportsHandler, swingStoreExportDir string) *types.GenesisState {
+func ExportGenesis(
+	ctx sdk.Context,
+	k Keeper,
+	swingStoreExportsHandler *SwingStoreExportsHandler,
+	swingStoreExportDir string,
+	swingStoreExportMode string,
+) *types.GenesisState {
 	gs := &types.GenesisState{
-		Params:               k.GetParams(ctx),
-		State:                k.GetState(ctx),
-		SwingStoreExportData: nil,
+		Params:                   k.GetParams(ctx),
+		State:                    k.GetState(ctx),
+		SwingStoreExportData:     nil,
+		SwingStoreExportDataHash: "",
```

Comment:
This should actually always be set, even in skip mode, to avoid triggering the ["bootstrap" logic from import genesis](https://github.com/Agoric/agoric-sdk/blob/4b1403aa06a6c08aae2d4cce0fd582480b7e3c7f/golang/cosmos/x/swingset/genesis.go#L44-L45).

Most likely we want to create the hasher in all cases (`hasher := sha256.New()`), and `gs.SwingStoreExportDataHash = fmt.Sprintf("sha256:%x", hasher.Sum(nil))` would be done even if we didn't call `InitiateExport`. The hasher will simply try to hash the empty string and return `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`.

Importing such a genesis [will cause a panic when trying to open the export directory](https://github.com/Agoric/agoric-sdk/blob/4b1403aa06a6c08aae2d4cce0fd582480b7e3c7f/golang/cosmos/x/swingset/genesis.go#L50-L53), which is what we want.

Reply:
I thought we want to trigger the bootstrap logic
And will the reason of the panic be the absence of directory pointed by `swingStoreExportDir`?

Reply:
We definitely don't want to trigger swingset bootstrap for any genesis export, at least not without a manual intervention on the genesis file. And yeah it'd be the missing directory as the genesis would be saying "hey look at the data alongside for swing set" and there is none.